internals/src/macros.rs

```diff
@@ -139,9 +139,12 @@ macro_rules! debug_from_display {
 /// Asserts a boolean expression at compile time.
 #[macro_export]
 macro_rules! const_assert {
-    ($x:expr) => {{
-        const _: [(); 0 - !$x as usize] = [];
-    }};
+    ($x:expr $(; $message:expr)?) => {
+        if !$x {
+            // We can't use formatting in const, only concating literals.
+            panic!(concat!("assertion ", stringify!($x), " failed" $(, $message)?))
```

Comment:
Oh, actually, this has no separator for message so it'd become "failedplatforms". I think that rather than adding it to the string above we should put additional `": ", ` after `$(,`.

Reply:
Done, and tested with

```rust

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn const_assert() {
        const_assert!(1 == 0; "this is the message");
    }
}
```

Which outputs:

```bash
---- macros::tests::const_assert stdout ----
thread 'macros::tests::const_assert' panicked at internals/src/macros.rs:212:9:
assertion 1 == 0 failed: this is the message
```

Reply:
This made me realize our macro is misleading. It's not actually a const assert unless you assign it into a `const`. It needs to expand to `const _: () = if ...`